app/src/main/java/com/khuttun/notificationnotes/NotificationMgr.java

```diff
@@ -23,36 +22,39 @@
 {
     private static final String CHANNEL_ID = "notes";
     private static final int GROUP_NOTIF_ID = -1000;
-
+    private static final int SUMMARY_NOTIF_ID = -2000;
+    private static final String GROUP_KEY = "com.khuttun.notificationnotes";
     private Context context;
-    private NotificationManager notificationManager;
-    private NotificationCompat.Builder notificationBuilder;
+    private NotificationManagerCompat notificationManager;
+    private boolean groupAPI = false;
 
     public NotificationMgr(Context context)
     {
         this.context = context;
-        this.notificationManager = (NotificationManager) this.context.getSystemService(Context.NOTIFICATION_SERVICE);
+        this.notificationManager = NotificationManagerCompat.from(this.context);
 
-        // NotificationChannel is required on Oreo and newer
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)
+        // Grouping notifications together so they are expandable is available on Nougat and newer
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N)
         {
-            this.notificationManager.createNotificationChannel(new NotificationChannel(
-                    CHANNEL_ID,
-                    this.context.getString(R.string.channel_name),
-                    NotificationManager.IMPORTANCE_LOW));
+            this.groupAPI = true;
         }
+    }
 
-        this.notificationBuilder = new NotificationCompat.Builder(this.context, CHANNEL_ID);
-        this.notificationBuilder.setSmallIcon(R.drawable.pen);
-        this.notificationBuilder.setOngoing(true);
-        this.notificationBuilder.setPriority(NotificationCompat.PRIORITY_LOW);
-        this.notificationBuilder.setContentIntent(PendingIntent
+    public NotificationCompat.Builder getNotificationBuilder()
```

Comment:
What's the reason for changing the builder to not be a member anymore?

Reply:
I have been unable to get the notification that shows how many notes there (the top notification) to work using the builder as a member variable.  In general, having a new builder each time makes the building of notifications a bit easier to reason about.